opencti-platform/opencti-front/src/private/components/common/containers/ContainerAddStixCoreObjects.jsx

```diff
@@ -94,18 +96,71 @@ const ContainerAddStixCoreObjects = (props) => {
     ],
   };
 
-  const [filters, setFilters] = useState(
-    targetStixCoreObjectTypes
-    && !(targetStixCoreObjectTypes.includes('Stix-Domain-Object') || targetStixCoreObjectTypes.includes('Stix-Cyber-Observable'))
-      ? targetEntityTypesFilterGroup
-      : emptyFilterGroup,
+  const isTypeDomainObject = (types) => {
+    return !types
+      || types.some((r) => stixDomainObjectTypes.indexOf(r) >= 0)
+      || (types.length === 1 && types[0] === 'Stix-Domain-Object');
+  };
+  const isTypeObservable = (types) => {
+    return !types
+      || types.some((r) => stixCyberObservableTypes.indexOf(r) >= 0)
+      || (types.length === 1 && types[0] === 'Stix-Cyber-Observable');
+  };
+
+  const resolveAvailableTypes = () => {
+    if (
+      targetStixCoreObjectTypes
+      && isTypeDomainObject(targetStixCoreObjectTypes)
+      && !isTypeObservable(targetStixCoreObjectTypes)
+    ) {
+      return 'Stix-Domain-Object';
+    }
+    if (
+      targetStixCoreObjectTypes
+      && isTypeObservable(targetStixCoreObjectTypes)
+      && !isTypeDomainObject(targetStixCoreObjectTypes)
+    ) {
+      return 'Stix-Cyber-Observable';
+    }
+    if (
+      !targetStixCoreObjectTypes
+      || (isTypeObservable(targetStixCoreObjectTypes)
+        && isTypeDomainObject(targetStixCoreObjectTypes))
+    ) {
+      return 'Stix-Core-Object';
+    }
+    return null;
+  };
+
+  const LOCAL_STORAGE_KEY = `container-${containerId}-add-objects`;
+  const { viewStorage, helpers, paginationOptions: addObjectsPaginationOptions } = usePaginationLocalStorage(
```

Comment:
What is the goal of the hook? The filters should be saved in local storage?

Reply:
Yes. But not only the filters but there are some other utilities.

Reply:
But we are not saving it in local storage here? Or at least we are not fetching what is in local storage when loading the page as if I refresh I don't see the previous filters I made

Reply:
Yes, you are correct. I'm unsure about the intention here. The saving in localStorage is functional, but upon closing the drawer, we reset the state and clear all filters.
https://github.com/OpenCTI-Platform/opencti/pull/6086/files/eb6c9427dcab8747dfda8053e566206e15ad277a#diff-c501c533b095ec5c7f5b484df27e33d102e40b4369f12d850a58a82b465ea31bR419-R421

I believe we're utilizing the useLocalStoragePagination for the infinite loader rather than for storing the filter in localStorage.

Reply:
In this history, we are using here a simple filter state. If we can confirm the behavior, I can change the code and use "useFilterState" instead.

Reply:
See with the product. We will keep the state. I remove the resetState function